src/forms/TimeInput.jsx

```diff
@@ -19,8 +50,91 @@ class TimeInput extends React.Component {
 	* @description called when the input changes, in turn calls the onChange
 	* 	handler prop, if there is one provided (eg supplied by redux-form or DateTimePicker)
 	*/
-	onChange(e) {
-		this.props.onChange && this.props.onChange(e);
+	onChange(partialState) {
+		if (this.props.onChange) {
+			const stateValues = {
+				...this.state,
+				...partialState
+			};
+
+			const v = `${formatDigits(stateValues.hours) % (this.props.is24Hr ? 24 : 12)}:${formatDigits(stateValues.minutes)}`;
+			this.props.onChange(v);
+		}
+	}
+
+	/**
+	* @function onTimeInputChange
+	* @param e Event Object
+	* @description called when the browser-native time input changes, in turn calls the onChange
+	* 	handler prop, if there is one provided (eg supplied by redux-form or DateTimePicker)
+	*/
+	onTimeInputChange(e) {
+		const { value } = e.target;
+		this.setState(() => ({ value: value }));
```

Comment:
Just curious - any particular reason for doing this instead of just `this.setState({ value: value });`?

Reply:
I got feedback in a PR a while back that said to do it this way, so that's how I've been doing it ever since. Either way works though :)

Reply:
the callback form of `setState` is now [the default syntax for React](https://reactjs.org/docs/react-component.html#setstate), and the React maintainers have suggested that it might end up being the only supported syntax in future versions of React because it enables some performance optimizations that apparently are more difficult with the object syntax. It's definitely a 'loose' recommendation to use the callback rather than the object arg form, but it's guaranteed future-proof in a way that the object form is not, and has a tiny benefit of making it clear that the state update will be done asynchronously - you can get yourself into trouble thinking about `setState` as a synchronous call, and the object form doesn't give you any indication that it's otherwise.

I would recommend `{ value }` instead of `{ value: value }`, however :)